### ISSUE TITLE
fatal error: ffi.h: No such file or directory

### ISSUE THREAD

**Issue by DmitryRomanenko** [geerlingguy/JJG-Ansible-Windows#28]
Tested on `config.vm.box = "geerlingguy/ubuntu1604"`

```
Running setup.py bdist_wheel for cryptography: finished with status 'error'
==> default:   Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-KiMAPG/cryptography/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmp5uiFmwpip-wheel- --python-tag cp27:
==> default:   c/_cffi_backend.c:15:17: fatal error: ffi.h: No such file or directory
==> default:   compilation terminated.
```

You need to install the development package as well.
`libffi-dev` + `libssl-dev` on Debian/Ubuntu, `libffi-devel` + something for openssl dev package on Redhat/Centos/Fedora.

**Comment by oxyc** [comment]
For reference, `paramiko` which we install has a hard dependency on `cryptography` which state:

> For Debian and Ubuntu, the following command will ensure that the required dependencies are installed:

> $ sudo apt-get install build-essential libssl-dev libffi-dev python-dev

> For Fedora and RHEL-derivatives, the following command will ensure that the required dependencies are installed:

> $ sudo yum install gcc libffi-devel python-devel openssl-devel

https://cryptography.io/en/latest/installation/

**Comment by geerlingguy** [comment]
This was fixed in PR #30

**Comment by ipeacocks** [comment]
For Alpine Linux: `libffi-dev`, `openssl-dev` and `python3-dev`
